# Week 2 - Day 1: Descriptive Statistics

## Objective
Compute and interpret descriptive statistics including:
- Mean
- Median
- Mode
- Variance
- Standard Deviation
- Interquartile Range (IQR)

Dataset: tested.csv

In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv("tested.csv")
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


In [3]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')

In [4]:
age = df["Age"].dropna()

print("Mean:", age.mean())
print("Median:", age.median())
print("Mode:", age.mode()[0])
print("Variance:", age.var())
print("Standard Deviation:", age.std())

q1 = age.quantile(0.25)
q3 = age.quantile(0.75)
iqr = q3 - q1

print("IQR:", iqr)

Mean: 30.272590361445783
Median: 27.0
Mode: 21.0
Variance: 201.10669538455943
Standard Deviation: 14.181209235624422
IQR: 18.0


In [5]:
print("Minimum Age:", age.min())
print("Maximum Age:", age.max())

Minimum Age: 0.17
Maximum Age: 76.0


## Which Measure Better Represents the Typical Age?

The **median** is a better measure of the typical age because the Age column contains a wide range of values (from 0.17 to 76 years). These extreme values can influence the mean, while the median is more robust and better represents the center of the data.

## Summary

The **Age** column has:

- Mean: **30.27**
- Median: **27.00**
- Mode: **21.00**
- Standard Deviation: **14.18**
- IQR: **18.00**

The ages are spread across a wide range, from **0.17** to **76** years. The median provides a better representation of a typical passenger's age because it is less affected by extreme values than the mean.